[4422] Persist content clipboard across refreshes

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/4422

aachavarria committed:
fix on getPasteItemFromPath

aachavarria committed:
doing refresh on itemPaste type CUT

rart commented on ui/app/src/components/Navigation/PathNavigator/PathNavigator.tsx:
```diff
@@ -216,16 +216,24 @@ export default function PathNavigator(props: WidgetProps) {
+                })
+              );
+            }
+          } else {
+            // payload.clipboard.type === 'CUT
+            const parentPath = getParentPath(payload.target);
+            if (parentPath === withoutIndex(state.currentPath)) {
+              dispatch(pathNavigatorRefresh({ id }));
+            }
           }
```
What happens if you try to paste-from-cut an item on a child? We may need a validation there since I don't think that's allowed.

rart commented on ui/app/src/components/Navigation/PathNavigator/PathNavigator.tsx:
```diff
@@ -216,16 +216,24 @@ export default function PathNavigator(props: WidgetProps) {
         }
         case folderCreated.type:
         case itemsPasted.type: {
```
I'm thinking we should always send/refresh the path to which you're pasting on. If you're there refresh, if you're not there, get the navigator to display that path on which you pasted the new item (same for cut or copy).

rart commented on ui/app/src/models/Item.ts:
```diff
@@ -99,7 +99,7 @@ export interface LegacyItem {
-  uri?: string;
-  children?: CopyItem[];
+export interface PasteItem {
+  path?: string;
```
🤦🏻‍♂️

rart commented on ui/app/src/state/epics/content.ts:
```diff
@@ -161,34 +154,16 @@ const content = [
+          map(({ items }) => {
             return batchActions([
-              emitSystemEvent(itemsPasted({ target: payload.path, resultingPaths })),
+              emitSystemEvent(itemsPasted({ target: payload.path, clipboard: state.content.clipboard })),
```
Isn't the state.clipboard empty by this point?

aachavarria replied:
no because we are calling unSetClipBoard after this, so in this point the state still contains the clipboard

rart commented on ui/app/src/utils/path.ts:
```diff
@@ -131,6 +131,31 @@ export function getIndividualPaths(path: string, rootPath?: string): string[] {
 }
 
+export function getPasteItemFromPath(sourcePath: string, paths: string[]): PasteItem {
+  sourcePath = withoutIndex(sourcePath);
```
Should this be done to all paths?

aachavarria replied:
already fixed

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4422

aachavarria committed:
Fixing issue with getPasteItemFromPath

aachavarria committed:
adding new copy/paste to legacy

aachavarria committed:
Fix issue on ItemCopied listener

aachavarria committed:
adding paste and duplicate to legacy;

aachavarria committed:
adding CUT to legacy;
TODO: Cut/Paste Validations;

aachavarria committed:
adding itemMenu to wcm-root-folder.js

rart commented on static-assets/themes/cstudioTheme/yui/assets/skin.css:
```diff
@@ -5644,7 +5644,8 @@ h2.yui-editor-skipheader, h3.yui-editor-skipheader {
 
-.ygtvitem {
+.ygtv-highlight0:hover {
+  background-color: #e2e2e2;
```
Use rgba(...). Try to mimic material. (We're going to require some theming work legacy side to support dark mode.)

rart commented on ui/app/src/components/SystemStatus/GlobalDialogManager.tsx:
```diff
@@ -399,6 +400,18 @@ function GlobalDialogManager() {
       {/* endregion */}
+
+      {/* region Item Menu */}
+      {state.itemMenu.open && (
```
This will disrupt animation. Why the if?

rart commented on ui/scss/src/temp.scss:
```diff
@@ -416,7 +416,21 @@ div#acn-active-content {
+      top: 0;
+      right: 0;
+      cursor: pointer;
+    }
```
Are these styles for the ellipsis? Perhaps better to name as such, then. E.g. item-menu-trigger

aachavarria committed:
Adding ItemMenu and events propagation to wcm-assets-folder.js and wcm-root-folder.js;
Adding new duplicate to wmc_content.ts

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4422

 Conflicts:
	ui/app/src/services/content.ts

aachavarria committed:
adding isValidCutPastePath validation to cut and paste;

aachavarria committed:
removing conditional

aachavarria committed:
adding unlockItem to itemMenu;

aachavarria committed:
PreviewSearchPanel.tsx fixing issue with count;
PreviewInPageInstancesPanel.tsx: Fixing issue when null contentType;
PreviewComponentsPanel.tsx: Fixing issue with pushToolsPanelPage title;

aachavarria committed:
highlighted color issue on dark theme;

aachavarria committed:
Fixing issues related to the Groovy Controller
